syntax = "proto3";

package benchimage;

service FilterGRPC {
  rpc GrayscaleFilter (ImageGRPC) returns (ImageGRPC) {}
  rpc InvertFilter (ImageGRPC) returns (ImageGRPC) {}
  //rpc CartoonFilter (ImageGRPC) returns (ImageGRPC) {}
  //rpc PencilFilter (ImageGRPC) returns (ImageGRPC) {}
  //rpc SobelFilter (ImageGRPC) returns (ImageGRPC) {}
}

message ImageGRPC {
  repeated bytes content = 1;
}
